posixlib/src/main/scala/scala/scalanative/posix/string.scala

```diff
@@ -1,8 +1,105 @@
 package scala.scalanative.posix
 
-import scalanative.unsafe.{extern, CInt, CString}
+import scala.scalanative.unsafe._
+import scala.scalanative.posix.sys.types
+
+// The Open Group Base Specifications Issue 7, 2018 edition
+
+/* A CX comment before a method indicates a POSIX extension to
+ * the ISO/IEEE C standard.
+ *
+ * An XSI comment before method indicates it is defined in
+ * extended POSIX X/Open System Interfaces, not base POSIX.
+ */
 
```

Comment:
```suggestion
/** POSIX string
 *
 * The Open Group Base Specifications Issue 7, 2018 edition
 *
 * A CX comment before a method indicates a POSIX extension to
 * the ISO/IEEE C standard.
 *
 * An XSI comment before method indicates it is defined in
 * extended POSIX X/Open System Interfaces, not base POSIX.
 */
```

Reply:
There is also package level documentation that can be used if you want to put these boilerplate versions and extension comments in one place and not repeat them at the top of each file. You can run doc and view the results in a browser.

Reply:
Eric,

Thank you for the suggestions.  Yes, the Before/End markers are somewhat ugly.  They are
in the original specification and I have found them to be incredibly useful in determining
what is CX or XSI and what is straight POSIX/IEEE.

Besides, I get paid by the line of code.

Reply:
I will make the scaladoc edit you suggested.  That looks like an improvement.

Reply:
I pinged you on the chat with some more info if you want to discuss.

Reply:
I am not sure about the blank line but I put some info in Discord to help out.

Reply:
Thank you. I'll switch over there.

For durable future reference, especially the part about how to generate the scaladoc for
rapid development.

```
@LeeTibbert I left some suggestions on your PR - I think you will like the scaladoc documentation ideas. Out current docs look like this - https://javadoc.io/doc/org.scala-native/posixlib_native0.4_2.13/latest/index.html - from the icon on the README.md
I think signal is the only one that has any scaladoc. https://javadoc.io/doc/org.scala-native/posixlib_native0.4_2.13/latest/scala/scalanative/posix/index.html
You could even link the posix to the header on the scala doc https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/string.h.html
We could also maybe figure out how to make the version a parameter in the scaladoc generation so if it changes - 9699919799
FYI - https://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html and https://docs.scala-lang.org/style/scaladoc.html
You can run sbt:scala-native> posixlib2_13/doc to generate for rapid development. Open index.html in a browser.
```

Also for future reference. The `index.html` in question seems to be left under the project root in
`posixlib/.2.13/target/scala-2.13/api`.

Of course, someday, all of this will be in the `Contributor's Guide`.

Reply:
> We could also maybe figure out how to make the version a parameter in the scaladoc generation so if it changes 9699919799

Changes in the reference POSIX version, say to 2023, will force manual edits in that file anyway and force a change
the document number `9699919799` anyway.

More elaborate scaladoc change are left as an opportunity for another day.